codersdk/users.go

```diff
@@ -299,6 +316,26 @@ func (c *Client) LoginWithPassword(ctx context.Context, req LoginWithPasswordReq
 	return resp, nil
 }
 
+// ConvertToOAuthLogin will send a request to convert the user from password
+// based authentication to oauth based. The response has the oauth state code
+// to use in the oauth flow.
+func (c *Client) ConvertToOAuthLogin(ctx context.Context, req ConvertLoginRequest) (OauthConversionResponse, error) {
```

Comment:
Should this be called `ConvertLoginType` instead?

Reply:
Eh, I don't think so.

The route is `convert-login`, but this specific request to the convert is for converting to oauth.

You could in theory convert `oidc` -> `password` and reuse the route, but we would probably change the argument payload. And I'd probably just make a new client func that is like `ConvertToPasswordLogin` with the limited arg set?

Reply:
I might argue that if the route doesn't match what we'd actually call this then something is wrong. We should make sure they both make sense IMO.

Reply:
Renamed